echopype/echodata/simrad.py

```diff
@@ -0,0 +1,233 @@
+"""
+Contains functions that are specific to Simrad echo sounders
+"""
+from typing import Optional, Tuple
+
+from .echodata import EchoData
+
+
+def _check_mode_input_without_data(
+    waveform_mode: str, encode_mode: str, pulse_compression: bool
+) -> None:
+    """
+    Checks that the ``waveform_mode`` and ``encode_mode`` have
+    the correct values and that the pair of inputs are valid, without
+    considering the actual data.
+
+    Parameters
+    ----------
+    waveform_mode: str
+        Type of transmit waveform
+    encode_mode: str
+        Type of encoded return echo data
+    pulse_compression: bool
+        States whether pulse compression should be used
+    """
+
+    if waveform_mode not in ["CW", "BB"]:
+        raise RuntimeError("The input waveform_mode must be either 'CW' or 'BB'!")
+
+    if encode_mode not in ["complex", "power"]:
+        raise RuntimeError("The input encode_mode must be either 'complex' or 'power'!")
+
+    # BB has complex data only, but CW can have complex or power data
+    if (waveform_mode == "BB") and (encode_mode == "power"):
+        raise RuntimeError("encode_mode='power' not allowed when waveform_mode='BB'!")
+
+    # make sure that we have BB and complex inputs, if pulse compression is selected
+    if pulse_compression and ((waveform_mode != "BB") or (encode_mode != "complex")):
+        raise RuntimeError(
+            "Pulse compression can only be used with "
+            "waveform_mode='BB' and encode_mode='complex'"
+        )
+
+
+def _check_mode_input_with_data_EK60(
```

Comment:
what do you think about using "against" -- `_check_mode_input_against_data_EK60`  and the same for `..._EK80`?

Reply:
As we have discussed via slack we will change `_check_mode_input_with_data_EK60` to `_retrieve_correct_beam_group_EK60` and similarly for EK80.